docs/DeveloperGuide.md

```diff
@@ -25,6 +40,159 @@
 |v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
 |v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
 
+### Main Menu
+This section describes the implementation of core main menu features.
+
+**API:** `AllOnUs.java`
+
+The menu is accessed through a call to static method main() of the AllOnUs class. 
+The Class Diagram below shows the overall structure of the application from the menu's point of view.
+The AllOnUs class and other classes that are coupled to the menu are therefore included.
+
+![](images/MenuFeaturesClassDiagram.png)
+
+Note: Exception classes are left out of this diagram that aims to show the core structure of the application. is*Command()
+method here refers to all methods of this form (more on this will be elaborated later in detail). 
+
+As illustrated in the Class Diagram above, the AllOnUs class only associates to one of each of the other classes.
```

Comment:
the explanations maybe a bit too long for the reader maybe could shorten the explanation